packages/@glimmer/runtime/lib/vm/append.ts

```diff
@@ -478,12 +478,21 @@ export default class VM<T> implements PublicVM {
     return result.value;
   }
 
+  recover() {
+    this.elementStack.panic();
+  }
+
   next(): IteratorResult<RenderResult> {
     let { env, program, updatingOpcodeStack, elementStack } = this;
     let opcode = this.inner.nextStatement();
     let result: IteratorResult<RenderResult>;
     if (opcode !== null) {
-      this.inner.evaluateOuter(opcode, this);
+      try {
```

Comment:
should we use the babel-debug-helpers to make it so DEBUG mode this catch doesn't happen?